include/seqan3/range/view/detail.hpp

```diff
@@ -0,0 +1,362 @@
+    {
+        return view_type{std::forward<urng_t>(urange)};
+    }
+};
+
+/*!\brief A class template template that make a view pipeable in a generic way [version for view functions that take
+ * arguments].
+ * \ingroup view
+ * \copydetails declare_view_functor_type_without_args
+ */
+template <template <typename> class view_type>
+struct declare_view_functor_type_with_args
+{
+    /*!\brief Enables function style usage of the view, forwards to the constructor of view_type.
+     * \tparam urng_t Type of the underlying range. view_type must be constructible with an argument of urng_t
+     * and arguments of type arg_types.
+     * \tparam arg_types The types of the arguments.
+     * \param urange The underlying range.
+     * \param args Further arguments forwarded to the constructor of view_type.
+     * \returns An object of type `view_type`.
+     *
+     * \details
+     *
+     * Given:
+     *
+     * ```cpp
+     * template <typename urng_t>
+     *     requires seqan3::input_range_concept<urng_t> // or more/stricter requirements
+     * struct view_foo
+     * {
+     *     // your implementation
+     *     // constructor takes an int argument and e.g. stores it
+     * };
+     *
+     * using foo_fn = declare_view_functor_type_with_args<view_foo>;
+     *
+     * namespace view
+     * {
+     * foo_fn foo;
+     * }
+     * ```
+     *
+     * This functor enables the function call syntax:
+     *
+     * ```cpp
+     * std::vector<int> container{1, 2, 3};
+     *
+     * auto v = view::foo(container, 7); // v is now of type view_foo<std::vector<int>
+     * ```
+     */
+    template <typename urng_t, typename ...arg_types>
+    auto operator()(urng_t && urange, arg_types && ...args) const
+        requires (sizeof...(arg_types) > 0) &&
+                 requires { view_type{std::declval<urng_t>(), std::declval<arg_types>()...}; }
+    {
+        return view_type{std::forward<urng_t>(urange), std::forward<arg_types>(args)...};
+    }
+
+    /*!\brief Binds operator() with a placeholder for range so that the operator() without `urange` argument works.
+     * \tparam urng_t Type of the underlying range. view_type must be constructible with an argument of urng_t
+     * and arguments of type arg_types.
+     * \tparam arg_types The types of the arguments.
+     *
+     * \details
+     *
+     * This replaces std::bind which doesn't work here because of the parent being template template and thus
+     * the type not being fully resolved. A lambda function wrapper could be used, but its type wouldn't be
+     * deducible which we need for the definition of the pipe operator.
+     */
+    template <typename ...arg_types>
+    struct auxiliary_functor_t
+    {
+        //!\brief Store the arguments
+        std::tuple<arg_types...> arguments;
+
+        //!\brief Helper function to unpack the tuple.
+        template <typename urng_t, size_t... Is>
+        auto explode(urng_t && urange, std::index_sequence<Is...> const &)
```

Comment:
Just minor thing. This function should be scoped in a protected region. 
Not really an exposable interface. 
The same holds for the `arguments` member.